lib/run/index.js

```diff
@@ -136,13 +136,19 @@ module.exports = function (options, callback) {
         }
 
         // sets entrypoint to execute if options.folder is specified.
-        if (options.folder) {
+        if (options.folder && !options.itemFromFolder) {
             entrypoint = { execute: options.folder };
 
             // uses `multipleIdOrName` lookupStrategy in case of multiple folders.
             _.isArray(entrypoint.execute) && (entrypoint.lookupStrategy = MULTIENTRY_LOOKUP_STRATEGY);
         }
 
+        if (options.itemFromFolder) {
+            if (options.collection.items) {
```

Comment:
This is a bit risky. Direct manipulation of collection SDK's internal members object leads to inconsistencies.